pkg/controller/activegate/activegate_update.go

```diff
@@ -14,15 +15,33 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func (r *ReconcileActiveGate) updatePods(
+type updateService interface {
+	FindOutdatedPods(
+		r *ReconcileActiveGate,
+		logger logr.Logger,
+		instance *dynatracev1alpha1.ActiveGate) ([]corev1.Pod, error)
+	IsLatest(logger logr.Logger, image string, imageID string, imagePullSecret *corev1.Secret) (bool, error)
+	UpdatePods(
+		r *ReconcileActiveGate,
+		pod *corev1.Pod,
+		instance *dynatracev1alpha1.ActiveGate,
+		secret *corev1.Secret) (*reconcile.Result, error)
+}
+
+type activeGateUpdateService struct{}
+
+func (updateService *activeGateUpdateService) UpdatePods(
+	r *ReconcileActiveGate,
 	pod *corev1.Pod,
 	instance *dynatracev1alpha1.ActiveGate,
 	secret *corev1.Secret) (*reconcile.Result, error) {
-	if !instance.Spec.DisableActivegateUpdate &&
+	if instance == nil {
+		return nil, fmt.Errorf("instance is nil")
+	} else if !instance.Spec.DisableActivegateUpdate &&
 		instance.Status.UpdatedTimestamp.Add(UpdateInterval).Before(time.Now()) {
 		log.Info("checking for outdated pods")
 		// Check if pods have latest activegate version
-		outdatedPods, err := r.findOutdatedPods(log, instance, isLatest)
+		outdatedPods, err := r.updateService.FindOutdatedPods(r, log, instance)
```

Comment:
Can't it call itself?

```suggestion
		outdatedPods, err := updateService.FindOutdatedPods(r, log, instance)
```

Reply:
The reconciler could have a different service than the calling one

Reply:
So that `activeGateUpdateService.UpdatePods()` can call `FindOutdatedPods()` from a `updateService` that is not itself? Is this done for testing?